src/components/expansion-panel/expansion-panel.ts

```diff
@@ -0,0 +1,239 @@
+  @property({ reflect: true, type: Boolean })
+  public disabled = false;
+
+  /** The indicator alignment of the expansion panel. */
+  @property({ reflect: true, attribute: 'indicator-alignment' })
+  public indicatorAlignment: 'start' | 'end' = 'start';
+
+  @query('[part~="header"]', true)
+  protected panelHeader!: HTMLElement;
+
+  private panelId!: string;
+
+  constructor() {
+    super();
+  }
+
+  public override connectedCallback() {
+    super.connectedCallback();
+    const id = this.getAttribute('id');
+    this.panelId! = id ? id : 'igc-expansion-panel-' + ++NEXT_ID;
+  }
+
+  private handleClicked(event: Event) {
+    if (this.disabled) {
+      event.preventDefault();
+      return;
+    }
+
+    const el = event.target as HTMLElement;
+
+    if (!el.matches(TABBABLE_SELECTORS)) {
+      this.panelHeader!.focus();
+    }
+
+    if (this.open) {
+      this.closeWithEvent();
+    } else {
+      this.openWithEvent();
+    }
+  }
+
+  private handleKeydown(event: KeyboardEvent) {
+    if (this.disabled) {
+      return;
+    }
+
+    switch (event.key.toLowerCase()) {
+      case 'arrowdown':
+      case 'down':
+        if (event.altKey) {
+          this.openWithEvent();
+        }
+        break;
+      case 'arrowup':
+      case 'up':
+        if (event.altKey) {
+          this.closeWithEvent();
+        }
+        break;
+      case 'enter':
+      case ' ':
+        this.open ? this.closeWithEvent() : this.openWithEvent();
+        break;
+    }
+  }
+
+  /**
+   * @private
+   * Opens the panel.
+   */
+  private openWithEvent() {
+    if (this.open) {
+      return;
+    }
+
+    const args = {
+      cancelable: true,
+      detail: this,
+    };
+
+    const allowed = this.emitEvent('igcOpening', args);
+
+    if (!allowed) {
+      return;
+    }
+
+    this.open = true;
+    this.emitEvent('igcOpened', { detail: this });
+  }
+
+  /**
+   * @private
+   * Closes the panel.
+   */
+  private closeWithEvent() {
+    if (!this.open) {
+      return;
+    }
+
+    const args = {
+      cancelable: true,
+      detail: this,
+    };
+
+    const allowed = this.emitEvent('igcClosing', args);
+
+    if (!allowed) {
+      return;
+    }
+
+    this.open = false;
+    this.emitEvent('igcClosed', { detail: this });
+  }
+
+  /** Toggles panel open state. */
+  public toggle(): void {
+    this.open = !this.open;
+  }
+
+  /** Hides the panel content. */
+  public hide(): void {
+    this.open = false;
+  }
+
+  /** Shows the panel content. */
+  public show(): void {
+    this.open = true;
+  }
+
+  private indicatorTemplate() {
+    return html`
+      <div part="indicator" aria-hidden="true">
+        <slot name="indicator">
+          <igc-icon
+            name=${this.open ? 'keyboard_arrow_up' : 'keyboard_arrow_down'}
+            collection="internal"
+          >
+          </igc-icon>
+        </slot>
+      </div>
+    `;
+  }
+
+  private headerTemplate() {
+    return html`
+      <div
```

Comment:
The ARIA design pattern wraps the button in a heading (must check if that's always reqruired), which this is missing. Also if we wrap it internally there's no control over the aria level which might be a bit of an issue.

Reply:
Additionally, in terms of accessibility - if there's no title/sub projected this will still render as an empty button that is still targetable by tab navigation which I'm not sure is ideal.

Speaking of, we used expansion panel in some scenarios without a header even and if such a scenario is also intended to be supported then the hardcoded header would not fit.

Tagging @sdimchevski as well for this and I'll look for the example without a header to add here so we can evaluate

Reply:
Okay took a bit to dig up since this is (no longer?) embedded in the docs, but the first card in this sample is using an expansion panel:
https://www.infragistics.com/angular-demos/layouts/card-sample-4
Point being, it doesn't have a header and it's toggled by some external button that either need to be positioned by the Card layout or under the panel as in this case. So my question is would this be a case we should consider supporting?

Reply:
@damyanpetev I don't know why this Card in the example is using an expansion panel. Expansion panels by design should have a header that acts as the control for expanding and collapsing. I believe you can have a card and implement your own collapsible mechanic through the use of button events but that's not in the scope of the expansion panel.